app/src/main/java/org/schabi/newpipe/database/Migrations.java

```diff
@@ -160,5 +161,14 @@ public void migrate(@NonNull final SupportSQLiteDatabase database) {
         }
     };
 
+    public static final Migration MIGRATION_3_4 = new Migration(DB_VER_3, DB_VER_4) {
+        @Override
+        public void migrate(@NonNull final SupportSQLiteDatabase database) {
+            database.execSQL(
+                    "ALTER TABLE streams ADD COLUMN uploader_url TEXT NOT NULL default ''"
```

Comment:
1. Does this design not duplicate the `uploader_url` over all the streams? Should it instead not be a it's own table and referenced here? One problem would be that currently `uploader` is also duplicated all over, so that would need to be migrated later on too or together here. 

---

2. I'm personally not a fan of empty default strings. Instead it should just be nullable and handled properly. For me an empty stream means that the url is an empty string, a null means that it does not exist

Reply:
On point 2, that shouldn't be too hard to check for null instead of empty.

Reply:
Point 2 fixed in https://github.com/TeamNewPipe/NewPipe/pull/6919/commits/21bcadeecbfa90e2bc66e77d3eb17163ae42ad55

Reply:
I think Point 1 should be a different pr? Will probably need another database migration.

Reply:
https://github.com/TeamNewPipe/NewPipe/issues/6924